frame/vault/src/lib.rs

```diff
@@ -777,20 +777,31 @@ pub mod pallet {
 			);
 
 			let to = Self::account_id(vault_id);
+
+			let lp = Self::do_calculate_lp_tokens_to_mint(vault_id, &vault, amount)?;
+
+			ensure!(
+				T::Currency::can_deposit(vault.lp_token_id, from, lp) ==
+					DepositConsequence::Success,
+				Error::<T>::MintFailed
+			);
```

Comment:
Probably not related to this PR but I guess this kind of checks are not worth to put anymore as we are using transactional macro for pretty much all extrinsics. This will probably become a default in upcomming polkadot upgrades also. This code is pretty much a legacy piece that was ensuring that we could do the transfer before doing the actual transfer :). Just so you are aware. We can remove all this `can_deposit` as the `transfer` will fail if impossible.

Reply:
Ah good to know! I'm using the same check for the option pallet, so can I safely remove them? Same thing for  `can_withdraw` I suppose.